### PR TITLE
Torchvision patch

### LINKED ISSUE #76791: torchvision failures with NVFuser
### 🐛 Describe the bug

There are two errors thrown in torchvision tests when nvfuser is enabled:

### first error

Error:
```
Traceback (most recent call last):
  File "/scratch/dberard/local/vision/test/test_models.py", line 768, in test_detection_model
    _check_jit_scriptable(model, ([x],), unwrapper=script_model_unwrapper.get(model_name, None), eager_out=out)
  File "/scratch/dberard/local/vision/test/test_models.py", line 140, in _check_jit_scriptable
    script_out = sm(*args)
  File "/scratch/dberard/local/pytorch/torch/nn/modules/module.py", line 1129, in _call_impl
    return forward_call(*input, **kwargs)
RuntimeError: The following operation failed in the TorchScript interpreter.
Traceback of TorchScript (most recent call last):
RuntimeError: profiled_ints.size() == input_ints.size() && std::equal( profiled_ints.begin(), profiled_ints.end(), input_ints.begin()) INTERNAL ASSERT FAILED at "../torch/csrc/jit/codegen/cuda/parser.cpp":3045, please report a bug to PyTorch. profiling ivalue doesn't support merge
```

Repro: from the torchvision directory, run `python -m pytest test/test_models.py -k cuda-fasterrcnn_mobilenet_v3_large_fpn`.

There are also a number of other `test_detection_model` tests that are failing, which can be tested with `python -m pytest test/test_models.py -k test_detection_model`

### second error

Error:
```
... (many similar errors) ...
________________________________________________________________________________________________________ TestColorJitter.test_color_jitter_all[3-cuda-9] ________________________________________________________________________________________________________
Traceback (most recent call last):
  File "/scratch/dberard/local/vision/test/test_transforms_tensor.py", line 186, in test_color_jitter_all
    _test_class_op(
  File "/scratch/dberard/local/vision/test/test_transforms_tensor.py", line 81, in _test_class_op
    transformed_tensor_script = scripted_fn(tensor)
  File "/scratch/dberard/local/pytorch/torch/nn/modules/module.py", line 1129, in _call_impl
    return forward_call(*input, **kwargs)
torch.jit.Error: The following operation failed in the TorchScript interpreter.
Traceback of TorchScript (most recent call last):
  File "/scratch/dberard/local/vision/torchvision/transforms/functional_tensor.py", line 65, in forward
    c = get_dimensions(img)[0]
    if c not in permitted:
        raise TypeError(f"Input image tensor permitted channel values are {permitted}, but found {c}")
        ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~ <--- HERE
builtins.TypeError: Input image tensor permitted channel values are [1, 3], but found 26

======================================================================================================================= warnings summary ========================================================================================================================
torchvision/transforms/functional_pil.py:243
  /scratch/dberard/local/vision/torchvision/transforms/functional_pil.py:243: DeprecationWarning: BILINEAR is deprecated and will be removed in Pillow 10 (2023-07-01). Use Resampling.BILINEAR instead.
    interpolation: int = Image.BILINEAR,

torchvision/transforms/functional_pil.py:317
  /scratch/dberard/local/vision/torchvision/transforms/functional_pil.py:317: DeprecationWarning: NEAREST is deprecated and will be removed in Pillow 10 (2023-07-01). Use Resampling.NEAREST or Dither.NONE instead.
    interpolation: int = Image.NEAREST,

torchvision/transforms/functional_pil.py:333
  /scratch/dberard/local/vision/torchvision/transforms/functional_pil.py:333: DeprecationWarning: NEAREST is deprecated and will be removed in Pillow 10 (2023-07-01). Use Resampling.NEAREST or Dither.NONE instead.
    interpolation: int = Image.NEAREST,

torchvision/transforms/functional_pil.py:350
  /scratch/dberard/local/vision/torchvision/transforms/functional_pil.py:350: DeprecationWarning: BICUBIC is deprecated and will be removed in Pillow 10 (2023-07-01). Use Resampling.BICUBIC instead.
    interpolation: int = Image.BICUBIC,

test/test_transforms_tensor.py::TestColorJitter::test_color_jitter_all[1-cpu-0]
  /scratch/dberard/local/pytorch/torch/nn/modules/module.py:1129: UserWarning: createConditionalConstant profile_node %189 : int = prim::profile_ivalue(%187)
   does not have profile information (Triggered internally at  ../torch/csrc/jit/codegen/cuda/graph_fuser.cpp:108.)
    return forward_call(*input, **kwargs)

test/test_transforms_tensor.py::TestColorJitter::test_color_jitter_all[1-cpu-0]
  /scratch/dberard/local/pytorch/torch/nn/modules/module.py:1129: UserWarning: createConditionalConstant profile_node %194 : int = prim::profile_ivalue(%dtype.1)
   does not have profile information (Triggered internally at  ../torch/csrc/jit/codegen/cuda/graph_fuser.cpp:108.)
    return forward_call(*input, **kwargs)

test/test_transforms_tensor.py::TestColorJitter::test_color_jitter_all[1-cpu-0]
  /scratch/dberard/local/pytorch/torch/nn/modules/module.py:1129: UserWarning: createConditionalConstant profile_node %282 : int = prim::profile_ivalue(%280)
   does not have profile information (Triggered internally at  ../torch/csrc/jit/codegen/cuda/graph_fuser.cpp:108.)
    return forward_call(*input, **kwargs)

test/test_transforms_tensor.py::TestColorJitter::test_color_jitter_all[1-cpu-0]
  /scratch/dberard/local/pytorch/torch/nn/modules/module.py:1129: UserWarning: createConditionalConstant profile_node %571 : int = prim::profile_ivalue(%orig_dtype.5)
   does not have profile information (Triggered internally at  ../torch/csrc/jit/codegen/cuda/graph_fuser.cpp:108.)
    return forward_call(*input, **kwargs)

test/test_transforms_tensor.py::TestColorJitter::test_color_jitter_all[1-cpu-0]
  /scratch/dberard/local/pytorch/torch/nn/modules/module.py:1129: UserWarning: createConditionalConstant profile_node %561 : int = prim::profile_ivalue(%559)
   does not have profile information (Triggered internally at  ../torch/csrc/jit/codegen/cuda/graph_fuser.cpp:108.)
    return forward_call(*input, **kwargs)

-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
==================================================================================================================== short test summary info ====================================================================================================================
FAILED test/test_transforms_tensor.py::TestColorJitter::test_color_jitter_all[3-cuda-1] - torch.jit.Error: The following operation failed in the TorchScript interpreter.
FAILED test/test_transforms_tensor.py::TestColorJitter::test_color_jitter_all[3-cuda-2] - torch.jit.Error: The following operation failed in the TorchScript interpreter.
FAILED test/test_transforms_tensor.py::TestColorJitter::test_color_jitter_all[3-cuda-3] - torch.jit.Error: The following operation failed in the TorchScript interpreter.
FAILED test/test_transforms_tensor.py::TestColorJitter::test_color_jitter_all[3-cuda-4] - torch.jit.Error: The following operation failed in the TorchScript interpreter.
FAILED test/test_transforms_tensor.py::TestColorJitter::test_color_jitter_all[3-cuda-5] - torch.jit.Error: The following operation failed in the TorchScript interpreter.
FAILED test/test_transforms_tensor.py::TestColorJitter::test_color_jitter_all[3-cuda-6] - torch.jit.Error: The following operation failed in the TorchScript interpreter.
FAILED test/test_transforms_tensor.py::TestColorJitter::test_color_jitter_all[3-cuda-7] - torch.jit.Error: The following operation failed in the TorchScript interpreter.
FAILED test/test_transforms_tensor.py::TestColorJitter::test_color_jitter_all[3-cuda-8] - torch.jit.Error: The following operation failed in the TorchScript interpreter.
FAILED test/test_transforms_tensor.py::TestColorJitter::test_color_jitter_all[3-cuda-9] - torch.jit.Error: The following operation failed in the TorchScript interpreter.
```

Repro: `python -m pytest test/test_transforms_tensor.py -k test_color_jitter_all`

### Versions

torchvision - main branch
pytorch - https://github.com/pytorch/pytorch/pull/76006
A100

cc @fmassa @vfdev-5 @pmeier

### PR DESCRIPTION
**Pull request by jjsjann123**
Fixes #76791 

Note that this is a hot patch so we get to run upstream tests. I'm doing proper fix in our local repo and will update upstream code once those are merged/reviewed.

### CODE REVIEW

File: torch/csrc/jit/codegen/cuda/parser.cpp [hunk 1]
```diff
@@ -1330,7 +1333,8 @@ class IrParser {
                 : nullptr;
 
             Val* out = clamp(operand, min, max);
-            value_map.emplace(node->output()->unique(), out);
+            value_map.emplace(
```
Comment (davidberard98) [inline torch/csrc/jit/codegen/cuda/parser.cpp:1336]: (not asking you to fix anything here, I'm just curious) - is it accurate that this is only necessary as part of the "real fix?" coming from csarofeen#1675? Or is this necessary for the changes below?

**Review by davidberard98 — APPROVED** [review]
LGTM!

**Comment by davidberard98** [comment]
@jjsjann123 actually it looks like test_permutation_preservation_edge_case_2 is failing, can you take a look at that?

**Comment by jjsjann123** [comment]
Oops, there's some code diff between our devel and upstream. Will patch that.

**Comment by jjsjann123** [comment]
errr.... messed up my cherry-pick yesterday.... Hopefully this time everything is good :crossed_fingers:

File: torch/csrc/jit/codegen/cuda/parser.cpp (hunk 1 above)
Comment (jjsjann123) [reply torch/csrc/jit/codegen/cuda/parser.cpp:1369]: The few places with the memory marked on outputs are the real fixes for format propagation. (where we generate output with wrong shapes).

Changes below for profiling information are coming from https://github.com/csarofeen/pytorch/pull/1677

Together they should fix the two issues coming from torchvision

**Comment by jjsjann123** [comment]
Failures doesn't look like it's relevant.
@pytorchbot merge this

**Pull request closed without merging**